internal/pkg/rpaas/manager.go

```diff
@@ -163,3 +166,108 @@ type CertificateData struct {
 	Certificate string `json:"certificate"`
 	Key         string `json:"key"`
 }
+
+type InstanceAddress struct {
+	Hostname string `json:"hostname,omitempty"`
+	Ip       string `json:"ip,omitempty"`
+}
+
+type InfoBuilder struct {
+	Address     *InstanceAddress                     `json:"address,omitempty"`
+	Replicas    *int32                               `json:"replicas,omitempty"`
+	Plan        string                               `json:"plan,omitempty"`
+	Locations   []v1alpha1.Location                  `json:"locations,omitempty"`
+	Service     *nginxv1alpha1.NginxService          `json:"service,omitempty"`
+	Autoscale   *v1alpha1.RpaasInstanceAutoscaleSpec `json:"autoscale,omitempty"`
+	Binds       []v1alpha1.Bind                      `json:"binds,omitempty"`
+	Team        string                               `json:"team,omitempty"`
+	Name        string                               `json:"name,omitempty"`
+	Description string                               `json:"description,omitempty"`
+	Tags        []string                             `json:"tags,omitempty"`
+}
+
+func NewInfoInstance(instance *v1alpha1.RpaasInstance) *InfoBuilder {
+	info := &InfoBuilder{
+		Replicas:  instance.Spec.Replicas,
+		Plan:      instance.Spec.PlanName,
+		Locations: instance.Spec.Locations,
+		Service:   instance.Spec.Service,
+		Autoscale: instance.Spec.Autoscale,
+		Binds:     instance.Spec.Binds,
+		Name:      instance.ObjectMeta.Name,
+	}
+
+	if desc, ok := instance.ObjectMeta.Annotations["description"]; ok {
+		info.Description = desc
+	}
+	info.parseTags(instance)
+
+	return info
+}
+
+func (i *InfoBuilder) SetAddress(ctx context.Context, manager RpaasManager, instanceName string) error {
+	address, err := manager.GetInstanceAddress(ctx, instanceName)
+	if err != nil {
+		return err
+	}
+	i.Address = &InstanceAddress{
+		Ip: address,
+	}
+	return nil
+}
+
+func (i *InfoBuilder) parseTags(instance *v1alpha1.RpaasInstance) {
+	re1 := regexp.MustCompile(`.*?,`)
+	re2 := regexp.MustCompile(`,\w*$`)
+	tags := re1.FindAll([]byte(instance.ObjectMeta.Annotations["tags"]), -1)
+	last_tag := re2.Find([]byte(instance.ObjectMeta.Annotations["tags"]))
+
+	var new_tags []string
+	for _, tag := range tags {
+		new_tags = append(new_tags, strings.TrimSuffix(string(tag), ","))
+	}
+	if last_tag != nil {
+		new_tags = append(new_tags, strings.TrimSuffix(string(last_tag), ","))
+	}
+
+	if len(new_tags) > 0 {
+		i.Tags = new_tags
+	}
```

Comment:
According to the code below, the annotation should be prefixed by `rpaas.extensions.tsuru.io/`, so you should get the tags from `rpaas.extensions.tsuru.io/tags` key.

https://github.com/tsuru/rpaas-operator/blob/5f7503b7ce9f00f873a2def2dda39dea79f970ec/internal/pkg/rpaas/k8s.go#L1318

Reply:
Why are you using so much regexp? Is the strings.Split not enough?